File: src/test/java/no/nav/pto_schema/utils/ValidationUtils.java
package no.nav.pto_schema.utils;

import no.nav.pto_schema.kafka.json.JsonSchemaLocator;
import org.everit.json.schema.Schema;
import org.json.JSONObject;
import org.json.JSONTokener;

import static no.nav.common.json.JsonUtils.toJson;

public class ValidationUtils {

    public static <T> void validate(T obj) {
        Schema schema = JsonSchemaLocator.getKafkaSchema(obj.getClass());
        schema.validate(new JSONObject(new JSONTokener(toJson(obj))));
    }

}

File: src/test/kotlin/no/nav/pto_schema/utils/ValidationUtils.kt
package no.nav.pto_schema.utils

import no.nav.common.json.JsonUtils
import no.nav.pto_schema.kafka.json.JsonSchemaLocator
import org.json.JSONObject
import org.json.JSONTokener

object ValidationUtils {

    fun validate(obj: Any) {
        val schema = JsonSchemaLocator.getKafkaSchema(obj.javaClass)
        schema.validate(JSONObject(JSONTokener(JsonUtils.toJson(obj))))
    }
}

